server/cache/resource_cache.go

```diff
@@ -0,0 +1,62 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/core/v1"
+)
+
+type ResourceCache struct {
+	ctx    context.Context
+	cache  Cache
```

Comment:
There are other examples in the code where we use the decorator pattern to add caching to an interface. Maybe use that?

Reply:
so thing is , ResourceCache is not actually a generic cache,
It is specific for K8s objects, currently Service accounts and secrets

And `Cache` object is for generic `get` and `put`

I feel in this case simple composition makes more sense since contracts are different for `ResourceCache` and `Cache`